### ISSUE TITLE
`assertFalse` can have better typing

### ISSUE THREAD

**Issue by kt3k** [denoland/deno_std#2170]
ref https://github.com/denoland/deno_std/pull/2093#discussion_r862336747

**Comment by iuioiua** [comment]
@kt3k, is this able to be closed?

**Comment by kt3k** [comment]
I think this can be still explored a bit more.

What @bcheidemann suggested was:

```ts
type Falsy = false | null | undefined | "" | 0;
export function assertFalse(expr: unknown, msg = ""): asserts expr is Falsy {
  // --snip--
}
```

I'm not 100% sure this is correct way of typing it though.

**Comment by iuioiua** [comment]
Ok, why not make it complete? Officially, falsy would be defined by the [following](https://developer.mozilla.org/en-US/docs/Glossary/Falsy):
```ts
type Falsy = false | 0 | -0 | 0n | "" | '' | `` | null | undefined | NaN;
```
If this sounds good. I'm happy to do this.

**Comment by kt3k** [comment]
@iuioiua Sounds good to me, but TypeScript seems regarding `0` and `-0` as the same (No type error with `const a: -0 = 0`) and also "", '', and `` as the same.

**Comment by iuioiua** [comment]
Ah, that makes sense. Thanks for the correction. So, it'd be:
```ts
type Falsy = false | 0 | 0n | "" | null | undefined | NaN;
```
I'll submit the PR.

**Comment by kt3k** [comment]
fixed by #2562
